proto/expr.proto

```diff
@@ -275,6 +277,14 @@ message ExprNode {
     PG_GET_INDEXDEF = 2400;
     COL_DESCRIPTION = 2401;
     PG_GET_VIEWDEF = 2402;
+
+    // EXTERNAL
+    ICEBERG_BUCKET = 2201;
+    ICEBERG_TRUNCATE = 2202;
+    ICEBERG_YEAR = 2203;
+    ICEBERG_MONTH = 2204;
+    ICEBERG_DAY = 2205;
+    ICEBERG_HOUR = 2206;
```

Comment:
It's a little surprising that this many expressions are needed

Reply:
review note: big picture is here `IcebergPartitionInfo::convert_to_expression` https://github.com/risingwavelabs/risingwave/pull/14135/files#diff-fca15fe3e7a30b34c86607302831aab3e5354163a89ccedc5d36a167d60f4b91R89-R191

Reply:
Can you explain a bit why a single function, maybe like `iceberg_partition_value` isn't enough?

Reply:
Oh, is it because the "input" isn't exprs, but iceberg "transform"s, so it's not easy to construct? Hmmm 🧐

Reply:
> Oh, is it because the "input" isn't exprs, but iceberg "transform"s, so it's not easy to construct? Hmmm 🧐

Yes. Another reason is that different transform function has different signatures.
e.g.
`bucket(input,n) -> i32`, input should be `int, long, decimal, date, time, timestamp, timestamptz, string, uuid, fixed, binary`.
`truncate(input,w) -> input`, input should be `int, long, decimal, string`.

Actually, we also can use a function like 🤔
`iceberg_transform(type, metadata, input) -> any` 
But then we need to add lots of check logic in a single function like following and this will make this function too complex I think.
```
fn iceberg_transform<T>(type:i32, metadata: T, input:..) {
  switch type {
    bucket => ,
    truncate => ,
  }
}
```

Reply:
Oh, I find a way to avoid this, let me have a try.🥵

Reply:
The new implementation have avoid it now.